main.go

```diff
@@ -5,40 +5,39 @@ import (
 	"strings"
 )
 
-var setOfRules map[int]func([]string) []string
+var setOfRules []Rule
 var orderOfRules []int
 
+type Rule struct {
+	Divisor int
+	Effect  func([]string) []string
+}
+
 func determineOutput(number int) []string {
 
-	setOfRules = make(map[int]func([]string) []string)
-	orderOfRules = make([]int, 0)
+	setOfRules = make([]Rule, 0)
 	output := make([]string, 0)
 	// Go maps do not preserve order of insertion so need to have an array with order of rules
```

Comment:
I'd just delete this comment, you don't need to document Go's behaviour - that's what the offical docs are for! But happy for you to leave it in for now if it's useful while you're learning :)